### PR TITLE
Simpler parser ast

### PR DESCRIPTION
**Pull request by Julow**
Hi!

As I explained in ocaml/odoc#355, `Model.Comment` is a bit hard to obtain and understand and too strict and lossy on the input.

Some of this issues could be solved by changing `Model.Comment` with a bit of work. But I think that in the future, this will make similar issues harder to solve as Odoc codebase and language change.

This PR simplifies `Parser.Ast` to make it closer to the concrete syntax and simpler by delaying some parsing to `semantics.ml`:

- References are represented as strings
- `Raw_markup` targets can be any string or none.
- Keep concrete syntax used for lists
- Keep whitespaces (Only when a `Space` element is generated)

This PR also removes some uses of `Model` types, as @lpw25 suggested, but not all yet.

This PR is supposed to follow ocaml/odoc#351 but if this approach is definitively rejected, it should be a good first step at improving `Model.Comment`.

What do you think?

Thanks!

### CODE REVIEW

File: src/parser/syntax.ml [hunk 1]
```diff
@@ -90,9 +90,9 @@ let rec inline_element
     fun input location next_token ->
 
   match next_token with
-  | `Space ->
+  | `Space ws ->
     junk input;
-    Location.at location `Space
+    Location.at location (`Space ws)
```
Comment (aantron) [inline src/parser/syntax.ml]: This looks like a good place for an `as`-binding to avoid an allocation, which is introduced by this diff.

**Comment by Julow** [comment]
Thanks for your review !

I cleaned the code but did not manage to fix the CI with 4.08 yet.
Yes, I'd like to reopen #351 if possible, I just rebased the branch. The stability should not be an issue for us.

**Comment by aantron** [comment]
Thanks! The 4.08 failure is not due to this PR, and is tracked in #367. It seems GitHub won't let me reopen #351, because the underlying branch has been rebased. Would you mind opening a new PR, or pinging me if you move the branch back to the last commit in #351 when it was open?

**Comment by aantron** [comment]
Thanks! The 4.08 failure is not due to this PR, and is tracked in #367. It seems GitHub won't let me reopen #351, because the underlying branch has been rebased. Would you mind opening a new PR, or pinging me if you move the branch back to the last commit in #351 when it was open?

**Comment by Julow** [comment]
Thanks for merging !
I changed the branch of #351.

**Pull request merged**